internal/controller/clientpool/clientpool.go

```diff
@@ -21,16 +22,35 @@ import (
 	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type AuthMode string
+
+const (
+	AuthModeTLS           AuthMode = "TLS"
+	AuthModeAPIKey        AuthMode = "API_KEY"
+	AuthModeNoCredentials AuthMode = "NO_CREDENTIALS"
+	// Add more auth modes here as they are supported
+)
+
 type ClientPoolKey struct {
-	HostPort        string
-	Namespace       string
-	MutualTLSSecret string // Include secret name in key to invalidate cache when the secret name changes
+	HostPort   string
+	Namespace  string
+	SecretName string   // Include secret name in key to invalidate cache when the secret name changes
```

Comment:
unlikely but what if the secret name stays the same, but the key name changes in the `SecretKeyConfigRef`?